ios/MaasNearbyTransit/Sources/MaasNearbyTransit/Components/NearbyTransitStopBadgeView.swift

```diff
@@ -0,0 +1,101 @@
+import SwiftUI
+import Swappable
+
+struct NearbyTransitStopBadgeView: View, Swappable {
+
+    var constants: Kotlin<NearbyTransitStopBadgeConstants> { Kotlin(NearbyTransitStopBadgeConstants()) }
+
+    public struct InputType {
+        public let schedules: [Schedule]
+    }
+    let input: InputType
+
+    public init(input: InputType) {
+        self.input = input
+    }
+
+    public init(schedules: [Schedule]) {
+        self.input = .init(schedules: schedules)
+    }
+
+    private var slicedFilteredSchedules: [Schedule] {
+        input.schedules.filterDuplicates(includeElement: { $0.transportId == $1.transportId })
+    }
+
+    var defaultBody: some View {
+        ZStack {
+            ForEach(slicedFilteredSchedules.indices) {
+                imageView(schedule: slicedFilteredSchedules[$0])
+                    .padding([.leading, .bottom], $0 != 0 ? CGFloat((-constants.offsetXY * CGFloat($0))) : 0)
+            }
+        }
+    }
+
+    private func imageView(schedule: Schedule) -> some View {
+        Image.from(image: UIImage(named: schedule.transportId))
+            .frame(width: constants.imageWidth, height: constants.imageHeight, alignment: .center)
+            .background(schedule.color.parseColor())
+            .foregroundColor(Color.white)
+    }
+}
+
+private extension Array {
+
+    func filterDuplicates(includeElement: (_ lhs:Element, _ rhs:Element) -> Bool) -> [Element] {
+        var results = [Element]()
+
+        forEach { (element) in
+            let existingElements = results.filter {
+                return includeElement(element, $0)
+            }
+            if existingElements.count == 0 {
+                results.append(element)
+            }
```

Comment:
You can use first instead of filter, because you need at least one to check for duplicates, you don't need to find all
```suggestion
            let existingElement = results.first { includeElement(element, $0) }
            if existingElement == nil {
                results.append(element)
            }
```